### DISCUSSION TITLE
Disable button while submitting

### DISCUSSION THREAD

**Discussion by aeropagz** [bigskysoftware/htmx#1787] [Q&A]
Is there a way to disable a button while a request is still inflight to prevent multiple requests to be fired?

**Comment by Telroshan** [comment]
If your goal is to prevent multiple requests at the same time, [hx-sync](https://htmx.org/attributes/hx-sync/) could be a good fit

If you really wish to toggle the `disabled` attribute on the button, you could bind some custom JS to htmx events, such as the [htmx:beforeSend](https://htmx.org/events/#htmx:beforeSend) and [htmx:afterRequest](https://htmx.org/events/#htmx:afterRequest) events

**Comment by matiboy** [comment]
You can also use the [loading-states extension](https://htmx.org/extensions/loading-states/)

```html
<div hx-ext="loading-states">

<button hx-get="/demo" hx-target="#outcome" data-loading-disable>
  Click me</button>
<div id="outcome"></div>
</div>

```

See it in action [in this codepen](https://codepen.io/matiboy/pen/bGOWggj)

**Comment by dmenisdev** [comment]
A bit late but you can try the `hx-disabled-elt` see HTMX docu: https://htmx.org/attributes/hx-disabled-elt/ .
